src/middlewared/middlewared/plugins/alert.py

```diff
@@ -519,12 +517,9 @@ async def __should_run_or_send_alerts(self):
             return False
 
         if (
-            not await self.middleware.call('system.is_freenas') and
+            await self.middleware.call('system.is_enterprise') and
             await self.middleware.call('failover.licensed') and
-            (
-                await self.middleware.call('failover.status') == 'BACKUP' or
-                await self.middleware.call('failover.in_progress')
-            )
+            await self.middleware.call('failover.status') != 'MASTER'
```

Comment:
What if it is `SINGLE`?

Also I wanted to do the same check but @william-gr had some serious objections against it, let's revise them once again.

Reply:
`SINGLE` is returned if there are:
1. no HA network interfaces configured or
2. there are no zpools imported

We shouldn't run alerts in those scenarios either since HA won't be functional. I'll add @william-gr on here too so he can review 👍

Reply:
The reason for that IIRC is that we have other statuses to account for.

The bottomline is that I am very agaisnt chaging things for the sake of change. If we have to change something it needs a very strong/undeniable argument, because much of our codebase has a lot of gotchas with uncommented code and no regression/integration/unit tests for any of it. ;)

Reply:
This is causing database corruption. The `system_alert` table is different between the controllers.

Reply:
What causes database corruption is the database sync routines, not how alerts are run

Reply:
Maybe they do, but not in this scenario. The `flush_alerts` method is run every 5 mins and deletes all entries in the `system_alert` table and then goes through all the alerts in memory and rewrites them but only does this `if failover.status == 'BACKUP'`. If failover.status == 'ERROR', then it will run.

Reply:
Waiting on more collateral internally before doing anything else. Will open fresh PR as needed.